zebra-consensus/src/checkpoint.rs

```diff
@@ -905,9 +905,12 @@ where
         });
 
         async move {
-            commit_finalized_block
-                .await
-                .expect("commit_finalized_block should not panic")
+            let fut = commit_finalized_block.await;
```

Comment:
Since you've awaited it, this isn't a future any more, it's just a result.
```suggestion
            let result = commit_finalized_block.await;
```